### PR TITLE
ACMS-1499: Fixes acquia_cms_common permissions

### PR DESCRIPTION
**Pull request by deepakmishra2**
**Motivation**

Fixes ACMS-1499

**Proposed changes**

Refactoring permission so that on install permission is assigned to those roles

**Alternatives considered**

**Testing steps**

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_common/acquia_cms_common.install [hunk 1]
```diff
@@ -158,4 +158,25 @@ function acquia_cms_common_content_model_role_presave_alter(RoleInterface &$role
         break;
     }
   }
+  if (\Drupal::moduleHandler()->moduleExists('captcha')) {
```
Comment (apathak18) [inline modules/acquia_cms_common/acquia_cms_common.install:161]: @deepakmishra2 can we do something like this?
 $modules_permission = [
    'shield' => 'administer shield',
    'honeypot' => 'administer honeypot',
    'captcha' => 'administer CAPTCHA settings',
    'recaptcha' => 'administer recaptcha',
  ]
  switch ($role->id()) {
    case 'user_administrator':
      foreach ($modules_permission as $module => $permission) {
        if (\Drupal::moduleHandler()->moduleExists($module)) {
          $role->grantPermission($permission);
        }
      }
    break;
  }

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.module [hunk 2]
```diff
@@ -174,15 +174,18 @@ function acquia_cms_site_studio_rebuild_styles() {
...
-    'content_administrator',
-    'content_editor',
-    'content_author',
-    'site_builder',
-  ])) {
-    $permissions = SiteStudioPermissionHelper::getSiteStudioPermissionsByRole($role->id());
-    foreach ($permissions as $permission) {
-      $role->grantPermission($permission);
+  $config = \Drupal::configFactory()->getEditable('cohesion_sync.cohesion_sync_package');
+  if ($config) {
```
Comment (apathak18) [inline modules/acquia_cms_site_studio/acquia_cms_site_studio.module:178]: @deepakmishra2  can we do something like this?
if ($config = \Drupal::configFactory()->getEditable('cohesion_sync.cohesion_sync_package')) {

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.module [hunk 3]
```diff
@@ -174,15 +174,18 @@ function acquia_cms_site_studio_rebuild_styles() {
...
+    if (in_array($role->id(), [
+      'content_administrator',
+      'content_editor',
+      'content_author',
+      'site_builder',
+    ])) {
+      $permissions = SiteStudioPermissionHelper::getSiteStudioPermissionsByRole($role->id());
+      foreach ($permissions as $permission) {
```
Comment (apathak18) [inline modules/acquia_cms_site_studio/acquia_cms_site_studio.module:186]: @deepakmishra2 Replace this with user_role_grant_permissions($role->id(), $permissions);

**Review by apathak18 — CHANGES_REQUESTED** [review]
@deepakmishra2 provided some suggestions please look into that.

File: modules/acquia_cms_common/acquia_cms_common.install (hunk 1 above)
Comment (deepakmishra2) [reply modules/acquia_cms_common/acquia_cms_common.install]: updated these changes

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.module (hunk 2 above)
Comment (deepakmishra2) [reply modules/acquia_cms_site_studio/acquia_cms_site_studio.module]: updated these changes

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.install [hunk 4]
```diff
@@ -228,3 +228,32 @@ function acquia_cms_site_studio_update_8004() {
...
+  $permissions = [
+    'access cpt_cat_general_components cohesion_component_category group',
+    'access cpt_cat_hero_components cohesion_component_category group',
+    'access cpt_cat_interactive_components cohesion_component_category group',
+    'access cpt_cat_media_components cohesion_component_category group',
+    'access cpt_cat_template_components cohesion_component_category group',
+    'access hlp_cat_general_helpers cohesion_helper_category group',
+    'access hlp_cat_interactive_helpers cohesion_helper_category group',
+  ];
+  foreach ($role_list as $roles) {
```
[the remark this answers is not in the document]
Comment (deepakmishra2) [reply modules/acquia_cms_site_studio/acquia_cms_site_studio.install]: updated these changes

File: modules/acquia_cms_site_studio/acquia_cms_site_studio.module (hunk 3 above)
Comment (deepakmishra2) [reply modules/acquia_cms_site_studio/acquia_cms_site_studio.module]: updated these changes

**Pull request closed without merging**